roofit/roofitcore/inc/RooFit/ModelConfig.h

```diff
@@ -309,7 +309,9 @@ class ModelConfig final : public TNamed, public RooWorkspaceHandle {
    /// alias for GetWS()
    RooWorkspace *GetWorkspace() const { return GetWS(); }
 
-   void GuessObsAndNuisance(const RooAbsData &data, bool printModelConfig = true);
+   void GuessObsAndNuisance(const RooArgSet &obsSet, bool printModelConfig = true);
+
+   inline void GuessObsAndNuisance(const RooDataSet &data, bool printModelConfig = true) { return GuessObsAndNuisance(data, printModelConfig); }
```

Comment:
There is a Clang warning for this function definition:
```
roofit/roofitcore/inc/RooFit/ModelConfig.h:314:90: warning: all paths through this function will call itself [-Winfinite-recursion]
  314 |    inline void GuessObsAndNuisance(const RooDataSet &data, bool printModelConfig = true) { return GuessObsAndNuisance(data, printModelConfig); }
      |                                                                                          ^
```

To me, it seems the compiler is correct - is this missing something? Quickly looking at the Doxygen, maybe `*data.get()` to get a `RooArgSet`?

Reply:
Ah yes, sure! Thanks a lot for the catch :+1: I kept this signature for backwards compatibility but didn't use it, hence I missed this.